src/coreclr/jit/lowerxarch.cpp

```diff
@@ -4575,6 +4602,16 @@ void Lowering::ContainCheckStoreIndir(GenTreeStoreInd* node)
         MakeSrcContained(node, src);
     }
 
+    // If the source is a BSWAP, contain it on supported hardware to generate a MOVBE.
+    if (comp->opts.OptimizationEnabled() && src->OperIs(GT_BSWAP, GT_BSWAP16) &&
+        comp->compOpportunisticallyDependsOn(InstructionSet_MOVBE) && IsSafeToContainMem(node, src))
```

Comment:
Sorry, I just realized that this is also missing a size check (similar to on the load side).
I think this will cause problems for e.g. `b[0] = (byte)ReverseEndianness(x);` where it will end up trying to generate a byte-sized `movbe` again.

Reply:
Ah, good catch. Yes, that's problematic.